flowmachine/flowmachine/features/subscriber/handset_stats.py

```diff
@@ -0,0 +1,167 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+
+valid_stats = {"count", "sum", "avg", "max", "min", "median", "stddev", "variance"}
+valid_characteristics = {
+    "width",
+    "height",
+    "depth",
+    "weight",
+    "display_width",
+    "display_height",
+}
+
+from ...core import Table
+from .metaclasses import SubscriberFeature
+from .subscriber_tacs import SubscriberHandsets
+
+
+class HandsetStats(SubscriberFeature):
+    """
+    This class calculates statistics associated with numeric fields of the TAC
+    table, such as width, height, etc.
+
+    A subscriber might use different phones for different periods of time. In
+    order to take the different usage periods into account we must calculate
+    weighted statistics of the desired characteristics. As such, here we
+    calculate the weighted characteristic, weighted by the number of seconds a
+    subscriber held that handset.  Given that we only learn about changes in
+    handset when an event occurs, this average will be biased.  For instance,
+    if a subscriber changes his handset straight after an event occur, we will
+    be tracking the handset that the subscriber used to handle the event and
+    not the handset he actually switched to. We will only learn about the new
+    handset once a new event occurs.
+
+    We further assume that the handset held before the first observed event in
+    the series is equal to the handset of the first event. Likewise, we assume
+    that the handset held after the last observed event in the series is the
+    handset of the last event.
+
+    Parameters
+    ----------
+    characteristic: {"width", "height", "depth", "weight", "display_width", "display_height"}
+        Numeric handset characteristics present in the TAC table.
+    statistic : {'count', 'sum', 'avg', 'max', 'min', 'median', 'mode', 'stddev', 'variance'}, default 'sum'
+        Defaults to sum, aggregation statistic over the durations.
+    subscriber_handsets: flowmachine.features.subscriber_tacs.SubscriberHandsets
+        An instance of SubscriberHandsets listing the handsets associated with
+        the subscribers during the period of observation.
+
+    Examples
+    --------
+
+    >>> s = HandsetStats("2016-01-01", "2016-01-08", "width")
```

Comment:
Example doesn't match the signature now.

Reply:
Done.